### PR TITLE
[SW-20] Refactor star wars movie list page to use redux

### PR DESCRIPTION
**Pull request by svIZsollei**

### CODE REVIEW

**Review by svekukucska — APPROVED** [review]

**Comment by spinenco-sv** [comment]
@svIZsollei Please resolve conflicts and PR link to trello

File: src/pages/StarWarsMovies/index.js [hunk 1]
```diff
@@ -41,4 +26,22 @@ class StarWarsMovies extends React.Component {
...
+const mapStateToProps = ({ movies: { movies } }) => ({
+  movies,
+});
+
+const mapDispatchToProps = (dispatch) => ({
+  getMovies: () => dispatch(fetchMovies()),
+});
+
+StarWarsMovies.propTypes = {
+  getMovies: propTypes.func,
```
Comment (spinenco-sv) [inline src/pages/StarWarsMovies/index.js]: please make this function required

File: src/actions/movies.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-    .then(({ results }) => {
-      dispatch(setMovies(results));
-    })
-    .catch(() => {
-      alert("Error: Movie List not received!");
-    });
-};
+export const setMoviesAction = (movies) => ({
+  type: 'TAKE_MOVIES',
+  results: movies,
```
Comment (spinenco-sv) [inline src/actions/movies.js]: you can rename the **movies** parameter to **results**

File: src/reducers/movies.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import TAKE_MOVIES from '../actions/types';
+
+const initialState = {
+  movies: [{ id: 1 }],
```
Comment (spinenco-sv) [inline src/reducers/movies.js]: Please remove { id: 1 }

**Review by spinenco-sv — CHANGES_REQUESTED** [review]

**Review by spinenco-sv — APPROVED** [review]

**Pull request merged**